src/software/gui/standalone_simulator/widgets/standalone_simulator_draw_function_visualizer.cpp

```diff
@@ -23,7 +26,9 @@ void StandaloneSimulatorDrawFunctionVisualizer::mousePressEvent(QMouseEvent* eve
     // If Ctrl is pressed, place the ball where the user clicks
     if (event->modifiers() & Qt::ControlModifier && standalone_simulator)
     {
+        ctrl_clicked         = true;
         Point point_in_scene = createPoint(mapToScene(event->pos()));
+        initial_point        = createPoint(mapToScene(event->pos()));
         standalone_simulator->setBallState(BallState(point_in_scene, Vector(0, 0)));
```

Comment:
can we reuse initial_point in setting ball state?

Reply:
Fixed